includes/Story_Post_Type.php

```diff
@@ -252,6 +265,79 @@ public function register_meta(): void {
 		);
 	}
 
+	/**
+	 * Filters whether a post has a post thumbnail.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param bool             $has_thumbnail true if the post has a post thumbnail, otherwise false.
+	 * @param int|WP_Post|null $post          Post ID or WP_Post object. Default is global `$post`.
+	 */
+	public function has_post_thumbnail( $has_thumbnail, $post ): bool {
+		$post = get_post( $post );
+		if ( $post && ! $has_thumbnail && $this->get_slug() === get_post_type( $post ) ) {
+			$poster        = get_post_meta( $post->ID, self::POSTER_META_KEY, true );
+			$has_thumbnail = (bool) $poster;
+		}
+
+		return $has_thumbnail;
+	}
+
+	/**
+	 * Filters the post thumbnail HTML to return a img of hotlink image.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param string                       $html              The post thumbnail HTML.
+	 * @param int                          $post_id           The post ID.
+	 * @param int                          $post_thumbnail_id The post thumbnail ID, or 0 if there isn't one.
+	 * @param string|int[]                 $size              Requested image size. Can be any registered image size name, or
+	 *                                                        an array of width and height values in pixels (in that order).
+	 * @param string|array<string, string> $attr              Query string or array of attributes.
+	 * @return string The post thumbnail image tag.
+	 */
+	public function post_thumbnail_html( $html, $post_id, $post_thumbnail_id, $size, $attr ): string {
+		$post = get_post( $post_id );
+		if ( ! $post || $html || $post_thumbnail_id || $this->get_slug() !== get_post_type( $post ) ) {
+			return $html;
+		}
+
+		/**
+		 * Poster.
+		 *
+		 * @var array{url?:string, width?: int, height?: int}|false $poster
+		 */
+		$poster = get_post_meta( $post->ID, self::POSTER_META_KEY, true );
+		if ( ! $poster ) {
+			return $html;
+		}
+
+		list( $width, $height ) = image_constrain_size_for_editor( (int) $poster['width'], (int) $poster['height'], $size );
+		$size_class             = \is_array( $size ) ? implode( 'x', $size ) : $size;
+
+		$default_attr = [
+			'src'      => $poster['url'],
+			'class'    => "attachment-$size_class size-$size_class",
+			'decoding' => 'async',
+			'loading'  => 'lazy',
+			'width'    => $width,
+			'height'   => $height,
+		];
+
+		$attr = wp_parse_args( $attr, $default_attr );
+
+		$attr = array_map( 'esc_attr', $attr );
+		$html = '<img';
+
+		foreach ( $attr as $name => $value ) {
+			$html .= " $name=" . '"' . $value . '"';
```

Comment:
Maybe use `esc_attr` on the value here?

Reply:
This is done 3 lines above.

Reply:
Hmm I see. It's not very obvious. Seems very easy to remove this by accident again in the future. Plus a bit wasteful and slow to loop over `$attr` twice.

I tend to prefer doing the `esc_attr` here instead of using `array_map`